lib/crawler/keyword/helpers/csv_parser.ex

```diff
@@ -0,0 +1,51 @@
+defmodule Crawler.Keyword.Helpers.CSVParser do
+  alias NimbleCSV.RFC4180, as: CSV
+
+  @keywords_file_max_length 1000
+  @keyword_min_length 1
+  @keyword_max_length 100
+
+  def keywords_limit, do: @keywords_file_max_length
+  def keyword_min_length, do: @keyword_min_length
+  def keyword_max_length, do: @keyword_max_length
+
+  def parse(file_path) do
+    keywords = parse_keyword_list_from_file(file_path)
+
+    case validate_keywords_length(keywords) do
+      {:ok, keywords} -> validate_keyword_length(keywords)
+      {:error, reason} -> {:error, reason}
+    end
+  end
+
+  defp parse_keyword_list_from_file(path) do
+    path
+    |> File.stream!()
+    |> CSV.parse_stream(skip_headers: false)
+    |> Enum.to_list()
+    |> List.flatten()
+  end
+
+  defp validate_keyword_length(keywords) do
+    keyword_length_exceeded_limit_fn = fn keyword ->
+      keyword_length = String.length(keyword)
+      keyword_length >= @keyword_max_length or keyword_length <= @keyword_min_length
+    end
+
+    case Enum.any?(keywords, keyword_length_exceeded_limit_fn) do
+      false -> generate_ok(keywords)
+      true -> generate_error(:keyword_length_exceeded)
```

Comment:
It would be good to show the user what exactly keyword is not correct in error message. Let's imagine that the user uploaded 100 keywords with about the same length and one of the keywords is a bit longer than the limit. It will be hard for the user to find out where the error is.

But I think we already delayed the review for this PR, so maybe it can be done in the next PRs if you like the idea, otherwise, just ignore this comment.

Reply:
@Nihisil noted. Thanks for pointing it out.